pulp_container/tests/functional/api/test_sync_signatures.py

```diff
@@ -0,0 +1,101 @@
+import pytest
+
+from pulp_smash.pulp3.bindings import monitor_task
+from pulp_smash.pulp3.utils import gen_repo
+
+from pulpcore.client.pulp_container import (
+    ContainerContainerRepository,
+    ContainerRepositorySyncURL,
+)
+from pulp_container.tests.functional.utils import gen_container_remote
+
+REDHAT_REGISTRY_V2 = "https://registry.access.redhat.com"
+DEPRECATED_REPOSITORY_NAME = "rhel7-rhel-minimal"
+SINGLE_MANIFEST_TAG = "7.9-511-source"
+MANIFEST_LIST_TAG = "7.9"
+SIGSTORE_URL = "https://access.redhat.com/webassets/docker/content/sigstore"
+
+
+@pytest.fixture
+def repository(
+    delete_orphans_pre,
+    container_repository_api,
+    container_remote_api,
+    gen_object_with_cleanup,
+    request,
+):
+    """A repository that contains signatures synced from sigstore if specified."""
+    data = gen_container_remote(
+        url=REDHAT_REGISTRY_V2,
+        upstream_name=DEPRECATED_REPOSITORY_NAME,
+        policy="on_demand",
+        include_tags=[MANIFEST_LIST_TAG, SINGLE_MANIFEST_TAG],
+    )
+
+    if request.param:
+        data["sigstore"] = request.param
+
+    remote = gen_object_with_cleanup(container_remote_api, data)
+
+    data = ContainerContainerRepository(**gen_repo())
+    repository = gen_object_with_cleanup(container_repository_api, data)
+
+    signed_only = True if request.param else False
```

Comment:
It does not mean that signed_only is flipped automatically to True if sigstore is provided. It can happen that not all content is signed so we sync all content and available signatures, unless signed_only is set to True then we sync only content that has signatures + its signatures.

Reply:
you can add a test though with no sigstoreurl and signed_only=true

Reply:
there is also one more test, where there is signatore url and signed_only=true but some content turns out not to be signed but i think in the deprecated repo all the content is signed..

Reply:
> It does not mean that signed_only is flipped automatically to True if sigstore is provided. It can happen that not all content is signed so we sync all content and available signatures, unless signed_only is set to True then we sync only content that has signatures + its signatures.
Yes, I know that it does not mean that. I wanted only to initialize a fixture based on the passed sigstore URL. I can update the fixture to consider more scenarios.

Reply:
and this is exactly what i am saying - the fixture right now is flipping the signed_only to true if url to sigstore is present :)